### PR TITLE
Remove mdns

### LINKED ISSUE #2443: Snap failing to build due to mdns-publisher
mdns-publisher is failing to build with the following error:

```
Cloning into '/root/parts/mdns-publisher/src'...
Fetching godeps...
cannot find package "golang.org/x/tools/go/vcs" in any of:
	/snap/go/8304/src/golang.org/x/tools/go/vcs (from $GOROOT)
	/root/parts/mdns-publisher/go/src/golang.org/x/tools/go/vcs (from $GOPATH)

go get -d github.com/rogpeppe/godeps
Failed to run 'go get -d github.com/rogpeppe/godeps' for 'mdns-publisher': Exited with code 1.
Verify that the part is using the correct parameters and try again.
Run the same command again with --debug to shell into the environment if you wish to introspect this failure.
Error: Process completed with exit code 2.
```

I detest that it's failing to build with zero changes from us. My proposal isn't to actually fix this: it's to **remove it entirely.**

So what does this component actually DO? It's broadcasts a `nextcloud.local` (or `<hostname>.local`) hostname via mdns, which allows for easy network discovery. It's a feature that has existed from the beginning, when the primary purpose of this snap was the Nextcloud Box, a hardware device based on Ubuntu Core, which was entirely headless. Using mdns meant that you didn't actually need to connect a monitor to it or otherwise determine its IP address before being able to access it. That worked reasonably well, but today I think it's an out-of-date concern. The Nextcloud Box no longer exists, and thus the use-case behind why this component exists in the first place no longer exists.

The downside is: I don't generally like removing things from the snap. It's entirely possible there are users out there that depend on this feature. But that seems unlikely at this point. And if we get bugs logged by people who need it, we can always solve the build failure and add it back in. It's a scream test, which I hate, but it seems warranted here given that this isn't the first time this component has suddenly ceased building.

Thoughts?

### PR DESCRIPTION
**Pull request by Pilzinsel64**
Removes all mdns references to solve the actual build error and fixes #2443.

Sqash and merge because I'm a pig!

### CODE REVIEW

**Comment by Pilzinsel64** [comment]
Does this go to `master` or `develop`? Would be needed in `master` ASAP to fix the daily but `develop` should be also fine if we merge #2442 soon.

**Comment by kyrofa** [comment]
Yeah essentially everything to develop, please. I'll just rebase #2242.

File: src/patches/mysql-support-compile-time-disabling-of-setpriority.patch
Comment (kyrofa) [inline src/patches/mysql-support-compile-time-disabling-of-setpriority.patch:1]: Good catch, this should have been removed with 0f534193a67b6c9d283b2271ba8567a876247365. However, it's easier to review and revert and in general maintain things if we keep PRs focused on one specific thing. Would you mind proposing the removal of this file in its own PR?

**Review by kyrofa — CHANGES_REQUESTED** [review]
Nice work. You got all the technical details, but would you mind also updating the README to remove mentions of mDNS and network discovery?

File: src/patches/mysql-support-compile-time-disabling-of-setpriority.patch
Comment (Pilzinsel64) [reply src/patches/mysql-support-compile-time-disabling-of-setpriority.patch:1]: True. I moved it to #2445

**Comment by Pilzinsel64** [comment]
> ... I (or you) can just rebase #2242.

Sure that you refered the right PR? 😅

> ... but would you mind also updating the README to remove mentions of mDNS and network discovery?

Done. Should be just the one line at the top.

**Review by scubamuc — APPROVED** [review]

**Review by kyrofa — APPROVED** [review]

**Comment by Pilzinsel64** [comment]
And now it's just one single commit - thanks @kyrofa 
Still not sure how you freely can edit the history. I always just "Squash and commit" on other repos, haha.

**Pull request merged**